huntsman/utils/config.py

```diff
@@ -144,3 +146,47 @@ def query_config_server(key=None, name='config_server'):
             
 #==============================================================================
 
+def load_device_config(key=None, config_files=None, logger=None, **kwargs):
+    '''
+    Load the device config from either the config server or local files.
+    
+    Parameters
+    ----------
+    key:
+        The key used to query the config server. Only used if config_files
+        is None.
+    config_files:
+        List of config file names. If None (default), use the config server.
+        
+    Returns
+    -------
+    dict:
+        The config dictionary.
+    '''
+    if logger is None:
+        logger = DummyLogger()
+        
+    #Load config from local files?
+    if config_files is not None:
+        logger.debug(f'Loading config from local file(s).')
+        try:
+            config = load_config(config_files, **kwargs)
+        except Exception as e:
+            logger.error(f'Unable to load local config file(s): {e}')
+            raise(e)
+    
+    #Load config from the config server?
+    else:
+        if key is None:
+            key = get_own_ip()
+        logger.debug(f'Loading remote config with key: {key}')
+        try:
+            config = query_config_server(key=key)
+        except Exception as e:
+            logger.error(f'Unable to load remote config: {e}')
+            raise(e)
+            
+    return config
```

Comment:
maybe simplify the above with:
```suggestion
    return config[key]
```